lib/rspec/rails/matchers/have_rendered.rb

```diff
@@ -9,18 +9,32 @@ def initialize(scope, expected, message = nil)
             @expected = Symbol === expected ? expected.to_s : expected
             @message = message
             @scope = scope
+            @redirect_is = nil
           end
 
           # @api private
           def matches?(*)
-            match_unless_raises ActiveSupport::TestCase::Assertion do
+            match_check = match_unless_raises ActiveSupport::TestCase::Assertion do
               @scope.assert_template expected, @message
             end
+            check_redirect unless match_check
+            match_check
+          end
+
+          def check_redirect
+            response = @scope.response
+            return unless response.respond_to?(:redirect?) && response.redirect?
+            @redirect_is = @scope.send(:normalize_argument_to_redirection, response.location)
           end
```

Comment:
When the YARD doc is added, can you include some details about this `normalize_argument_to_redirection` method and why we need to use `send`?